devenv/dev-dashboards/gen.go

```diff
@@ -0,0 +1,93 @@
+//go:build ignore
+// +build ignore
+
+package main
+
+import (
+	"bytes"
+	"embed"
+	"os"
+	"path"
+	"strings"
+	"text/template"
+)
+
+//go:generate go run gen.go
+
+//go:embed *.json */*.json
+var devDashboardFS embed.FS
+
+//go:embed tmpl/*.tmpl
+var tmplFS embed.FS
+
+func main() {
+	g := newGen()
+	out, err := g.Generate()
+	if err != nil {
+		panic(err)
+	}
+	os.WriteFile("./gen.libsonnet", []byte(out), 0644)
```

Comment:
Really glad to see that some of the patterns in the other generators were helpful!

Right here is a place that's a really important one to adopt, i think - use of the diffwriter. Example from [plugins](https://github.com/grafana/grafana/blob/main/public/app/plugins/gen.go#L123-L135). This is what'll make it easy to check in CI that the generated code is up to date. (though you'd also need to ensure your make task is run in the existing CI step)

Reply:
> Really glad to see that some of the patterns in the other generators were helpful!

for sure, the use of `text/template` and `embed` were very helpful when the generation got too complex for a bash script

> generated code is up to date

ah ok, i was using `make` to manage conditional builds by including the dashboard JSON files in the  `gen.libsonnet` target dependencies. i'll look into `diffwriter`. thanks!

Reply:
`WriteDiffer` was added in https://github.com/grafana/grafana/pull/57928/commits/1bcb27fa8c416b5aa9f35d6100becd3e730c8787